src/main/resources/admin/tools/content-type-manager/components/component-file-handler.html

```diff
@@ -39,28 +48,72 @@
                 icon: "warning",
                 text: `Please select the content-types folder, you've selected a folder called ${dirHandle.name}.`,
               });
-              return;
+              return {};
             }
 
-            const { value: contentTypeName } = await Swal.fire({
-              text: "What is the name of the content type?",
-              input: "text",
-            });
+            return dirHandle;
+          };
+
+          // 3 - Check if the name of the new content type is valid
+          const checkIfContentTypeNameIsValid = async (dirHandle) => {
+            if (!("name" in dirHandle)) {
+              return { nameIsValid: false, ctyName: "" };
+            }
+
+            let ctyName = "";
+            const existingCtyNames = [];
 
-            if (!contentTypeName) return;
+            // Get all existing content type names
+            for await (const entry of dirHandle.values()) {
+              entry.kind === "directory" && existingCtyNames.push(entry.name);
+            }
+
+            const isCtyNameValid = (ctyName, existingCtyNames) =>
+              ctyName.length > 0 &&
+              !ctyName.includes(" ") &&
+              existingCtyNames.indexOf(ctyName) === -1;
+
+            const dialogText = (ctyName) => {
+              return ctyName
+                ? `${ctyName} is invalid or is the name of an existing content type. 
+                  Please define another one.`
+                : "What is the name of the content type?";
+            };
+
+            do {
+              const text = dialogText(ctyName);
+              const r = await Swal.fire({ text, input: "text" });
+              ctyName = r.value;
+            } while (!isCtyNameValid(ctyName, existingCtyNames));
+
+            return { nameIsValid: true, ctyName };
+          };
+
+          // 4 - Create the directory and the xml files for the content type
+          const createDirAndXmlFile = async ({ nameIsValid, ctyName }) => {
+            if (!nameIsValid) return false;
 
             const newContentTypeDirectoryHandle =
-              await dirHandle.getDirectoryHandle(contentTypeName, {
+              await dirHandle.getDirectoryHandle(ctyName, {
                 create: true,
               });
 
             this.fileHandle = await newContentTypeDirectoryHandle.getFileHandle(
-              `${contentTypeName}.xml`,
+              `${ctyName}.xml`,
               { create: true }
             );
 
-            this.$store.commit("setFileName", { data: contentTypeName });
-          });
+            this.$store.commit("setFileName", { data: ctyName });
+
+            return true;
+          };
+
+          R.pipeWith(R.andThen)([
```

Comment:
None of your code is functional programming. 
I don't think its good to mix programming styles. 
Then we would have to refactor the entire application to that style. 
Could you do something simular with just async await functions alone?